appveyor.yml

```diff
@@ -58,6 +68,53 @@ on_failure:
 artifacts:
   - path: dist\*
 
+# scripts to run before deployment
+before_deploy:
+  # install gravitybee/generate the standalone executable
+  - mkdir C:\venv
+  - C:\Python36-x64\Scripts\virtualenv C:\venv
+  - C:\venv\Scripts\activate
+  - python --version
+  - pip install gravitybee satsuki
+  - pip install --editable .
+  - gravitybee --src-dir src --sha file --extra-data static --verbose
+  # prepare deployment
+  - set SATS_TAG=$gb_pkg_ver
+  - set SATS_BODY=* [$gb_pkg_name v$gb_pkg_ver CHANGELOG](https://github.com/plus3it/$gb_pkg_name_lower/blob/$gb_pkg_ver/CHANGELOG.md) (Win)
+  - set SATS_REL_NAME=Release v$gb_pkg_ver
+  - set LOCAL_STAGING_DIR=s3_uploads
+  - ps: >-
+      if($env:APPVEYOR_REPO_BRANCH -eq "master") {
+        $env:S3_BUCKET=$env:RELEASE_BUCKET
+      } else {
+        $env:S3_BUCKET=$env:DEV_BUCKET
+      }
+  - set S3_KEYFIX=releases
+  - gravitybee-environs.bat
+  # upload & rename artifacts to appveyor's magic artifact depot in the sky (DITS)
+  - appveyor PushArtifact %GB_ENV_GEN_FILE_W_PATH% -FileName %GB_ENV_APP_VERSION%\%GB_ENV_GEN_FILE%
+  - appveyor PushArtifact %GB_ENV_APP_NAME%-%GB_ENV_APP_VERSION%-sha256-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.json -FileName %GB_ENV_APP_VERSION%\%GB_ENV_APP_NAME%-%GB_ENV_APP_VERSION%-sha256-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.json
+  - appveyor PushArtifact %GB_ENV_GEN_FILE_W_PATH% -FileName latest\%GB_ENV_APP_NAME%-latest-standalone-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.exe
+  - appveyor PushArtifact %GB_ENV_APP_NAME%-%GB_ENV_APP_VERSION%-sha256-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.json -FileName latest\%GB_ENV_APP_NAME%-latest-sha256-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.json
+  - SET
+  - ps: ls
+  - ps: >-
+      if($env:APPVEYOR_REPO_BRANCH -eq "master") {
+        satsuki --verbose
```

Comment:
does appveyor not have an equivalent of travis's "script" deploy provider?

Reply:
Nope. It is listed in the docs but is not a link like the other providers. I tried using it to test but it says it is not recognized.

Reply:
Actually this is not entirely true. AppVeyor has a stage/step called `deploy_script:` but it seems to be mutually exclusive with the `deploy:` stage/step. (More research needed.)